### PR TITLE
scheduler_v2: moveTableManager

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
- We need a module for implementing logic related to manual table moves.

### What is changed and how it works?
- Implemented `moveTableManager"

### Check List 

Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/scheduler/move_table_manager.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+			job.status = moveTableJobStatusRemoved
+			continue
+		case removeTableResultGiveUp:
+			delete(m.moveTableJobs, tableID)
+			// Giving up means that we can move forward,
+			// so there is no need to return false here.
+			continue
+		case removeTableResultUnavailable:
+		}
+		return false, nil
```
Comment (nongfushanquan) [inline cdc/scheduler/move_table_manager.go:145]: how to handle the return false case

File: cdc/scheduler/move_table_manager.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+func (m *moveTableManagerImpl) Add(tableID model.TableID, target model.CaptureID) bool {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	if _, ok := m.moveTableJobs[tableID]; ok {
+		// Returns false if the table is already in a move table job.
+		return false
+	}
+
+	m.moveTableJobs[tableID] = &moveTableJob{
```
Comment (nongfushanquan) [inline cdc/scheduler/move_table_manager.go:108]: why not  use the struct like  map[tableID][{captureID,status}]

**Review by 3AceShowHand — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Pull request merged**